packages/core/src/RenderingEngine/RenderingEngine.ts

```diff
@@ -852,7 +852,7 @@ class RenderingEngine implements IRenderingEngine {
     const viewport = new ViewportType(viewportInput);
 
     // 5. Storing the viewports
-    this._viewports.set(viewportId, viewport);
+    this._viewports.set(viewportId, viewport as IViewport);
```

Comment:
If you redeclare const viewportTypeToViewportClass as the code below, then you don't need the caset here:
// TODO -> Eventually we'll need to register to this list
import StackViewport from '../StackViewport';
import VolumeViewport from '../VolumeViewport';
import ViewportType from '../../enums/ViewportType';
import VolumeViewport3D from '../VolumeViewport3D';
import VideoViewport from '../VideoViewport';
import WSIViewport from '../WSIViewport';
import { ViewportInput, IViewport } from '../../types/IViewport';

interface ViewportConstructor {
  new (viewportInput: ViewportInput): IViewport;
}

const viewportTypeToViewportClass: {
  [key: string]: ViewportConstructor;
} = {
  [ViewportType.ORTHOGRAPHIC]: VolumeViewport,
  [ViewportType.PERSPECTIVE]: VolumeViewport,
  [ViewportType.STACK]: StackViewport,
  [ViewportType.VOLUME_3D]: VolumeViewport3D,
  [ViewportType.VIDEO]: VideoViewport,
  [ViewportType.WholeSlide]: WSIViewport,
};

export default viewportTypeToViewportClass;